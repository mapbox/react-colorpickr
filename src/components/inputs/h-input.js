'use strict';

import React from 'react';
import PropTypes from 'prop-types';
import NumberInput from './number-input';

class HInput extends React.Component {
  static propTypes = {
    theme: PropTypes.object.isRequired,
    label: PropTypes.string.isRequired,
    value: PropTypes.number.isRequired,
    onChange: PropTypes.func.isRequired
  };

  render() {
    return <NumberInput {...this.props} min={0} max={359} />;
  }
}

export default HInput;
