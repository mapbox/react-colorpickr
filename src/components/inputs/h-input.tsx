import React from 'react';
import { NumberInput } from './number-input.tsx';

function HInput(props) {
  return <NumberInput {...props} min={0} max={360} />;
}

export { HInput };
