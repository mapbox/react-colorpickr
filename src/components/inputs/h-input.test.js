import React from 'react';
import { render } from '@testing-library/react';
import { HInput } from './h-input';

describe('HInput', () => {
  test('renders', () => {
    const props = {
      id: 'h',
      value: 200,
      theme: {},
      onChange: jest.fn()
    };

    const { baseElement } = render(<HInput {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
