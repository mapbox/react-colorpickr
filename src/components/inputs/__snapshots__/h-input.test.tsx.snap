// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HInput renders 1`] = `
<body>
  <div>
    <div>
      <label
        for="h"
      >
        h
      </label>
      <input
        id="h"
        max="360"
        min="0"
        step="1"
        type="number"
        value="200"
      />
    </div>
  </div>
</body>
`;
